ow_plexil/src/plexil-adapter/OwInterface.cpp

```diff
@@ -804,6 +859,44 @@ void OwInterface::guardedMoveAction (double x, double y, double z,
     guarded_move_done_cb<GuardedMove, ow_lander::GuardedMoveResultConstPtr>);
 }
 
+std::vector<double> OwInterface::identifySampleLocation (int num_images, std::string filter_type, int id)
+{ 
+  m_sample_point.clear();
+  got_sample_location_result = false;
+  if (! mark_operation_running (Op_IdentifySampleLocation, id)) return m_sample_point;
+  thread action_thread (&OwInterface::identifySampleLocationAction, this, num_images, filter_type, id);
+  action_thread.detach();
+
+  ros::Rate rate(10);
+  int timeout = 0;
+  /* We wait for a result from the action server or the 5 sec timeout before returning
+  I dont think a timeout is strictly necessary here but because this is blocking
+  I put it in for safety */
+  while(got_sample_location_result == false && timeout < 50){
+      ros::spinOnce();
+      rate.sleep();
+      timeout+=1;
+  }
+  if(timeout == 50){
```

Comment:
Same as earlier comment, use a named constant for 50.   Also, good to embed the unit in the name, e.g. SampleTimeoutSeconds.